sdks/python/apache_beam/typehints/typecheck.py

```diff
@@ -146,9 +148,16 @@ def _type_check_result(self, transform_results):
       return transform_results
 
     def type_check_output(o):
-      # TODO(robertwb): Multi-output.
-      x = o.value if isinstance(o, (TaggedOutput, WindowedValue)) else o
-      self.type_check(self._output_type_hint, x, is_input=False)
+      if isinstance(o, TimestampedValue) and hasattr(o, "__orig_class__"):
```

Comment:
Please document what case the hasattr `__orig_class__` is checking. (is this for changes in typehint support between python versions? something else?)